src/commands/githubActions/bumpUpstream/setup.ts

```diff
@@ -0,0 +1,99 @@
+import { Manifest, Compose } from "@dappnode/types";
+import { readManifest, readCompose } from "../../../files/index.js";
+import { arrIsUnique } from "../../../utils/array.js";
+import { parseCsv } from "../../../utils/csv.js";
+import { readBuildSdkEnvFileNotThrow } from "../../../utils/readBuildSdkEnv.js";
+import { GitSettings, InitialSetupData, UpstreamSettings } from "./types.js";
+import { getFirstAvailableEthProvider } from "../../../utils/tryEthProviders.js";
+
+export async function readInitialSetup({ dir, userEthProvider, useFallback }: { dir: string, userEthProvider: string, useFallback: boolean }): Promise<InitialSetupData> {
+    const envFileArgs = readBuildSdkEnvFileNotThrow(dir);
+
+    const { manifest, format } = readManifest({ dir });
+    const compose = readCompose({ dir });
+
+    const upstreamRepos = envFileArgs
+        ? [envFileArgs._BUILD_UPSTREAM_REPO]
+        : parseCsv(manifest.upstreamRepo);
+    const upstreamArgs = envFileArgs
+        ? [envFileArgs._BUILD_UPSTREAM_VERSION]
+        : parseCsv(manifest.upstreamArg || "UPSTREAM_VERSION");
+
+    // Create upstream settings after validation
+    validateUpstreamData(upstreamRepos, upstreamArgs);
+    const upstreamSettings = upstreamRepos.map((repo, i) => ({
+        upstreamRepo: repo,
+        upstreamArg: upstreamArgs[i],
+    }));
+
+    const gitSettings = getGitSettings();
+
+    const ethProviders = getEthProviders(useFallback, userEthProvider);
+
+    const ethProviderAvailable = await getFirstAvailableEthProvider({
+        providers: ethProviders
+    });
+
+    if (!ethProviderAvailable)
+        throw Error(`No eth provider available. Tried: ${ethProviders.join(", ")}`);
+
+    return {
```

Comment:
For me it looks too arbitrary this data return. I would rather try to organize it by topics, a module like:
```
| settings
| -- index.js
| -- ethProvider
|  -- | -- index.js
|  -- | -- getEthProviders.js
| -- dappnodeFiles
|  -- | -- index.js
|  -- | -- getManifest.js
|  -- | -- getCompose.js
| -- upstream
|  -- | -- index.js
|  -- | -- validateUpstreamData.js
|  -- | -- getUpstreamData.js
| -- git
|  -- | -- index.js
```

Reply:
I did it in a simpler way, but I implemented it in module format